source/contentscript.js

```diff
@@ -1,11 +1,14 @@
-// when asked tell the extension about the SPDY/HTTP2 status of current page
-
+// determineConnectionInfo returns the protocol used to connect to the first hop
+// for the first navigation event available.
 function determineConnectionInfo() {
-  var loadTimes = chrome.loadTimes();
-  return {
-    spdy: loadTimes.wasFetchedViaSpdy,
-    type: loadTimes.npnNegotiatedProtocol || loadTimes.connectionInfo
-  };
+  if (performance && performance.getEntriesByType) {
+    let entry = performance.getEntriesByType('navigation')[0];
```

Comment:
Was wondering what this would hold in the case of a redirect and found something strange. It seems to represent the page/request after the redirect is complete but the redirectCount is still 0 and the name is set to the pre-redirect URL. There isn't a second entry to show the path from the URL that was first hit and then where it landed. Not a blocker but something to keep to keep in mind.

See this [screenshot](https://cloudup.com/cspAQDbfza7) for the case where I typed in `http://cloudflare.com` into the URL bar (do they still call this thing the omni bar?) and got redirected to `https://www.cloudflare.com` with a `307`